ui/src/app/views/new-notebook-modal/component.ts

```diff
@@ -0,0 +1,89 @@
+import {Component, Input, OnDestroy} from '@angular/core';
+
+import {SignInService} from 'app/services/sign-in.service';
+import {isBlank} from 'app/utils/index';
+import {Kernels} from 'app/utils/notebook-kernels';
+
+import {environment} from 'environments/environment';
+
+import {Workspace} from 'generated';
+
+
+@Component({
+  selector: 'app-new-notebook-modal',
+  styleUrls: ['./component.css',
+    '../../styles/buttons.css',
+    '../../styles/inputs.css'],
+  templateUrl: './component.html',
+})
+export class NewNotebookModalComponent implements OnDestroy {
+  public creatingNotebook = false;
+  public newName = '';
+  @Input() workspace: Workspace;
+  Kernels = Kernels;
+  kernelType: number = Kernels.Python3;
+  notebookAuthListeners: EventListenerOrEventListenerObject[] = [];
+
+  loading = false;
+
+  constructor(
+    private signInService: SignInService,
+  ) {}
+
+  ngOnDestroy(): void {
+    this.notebookAuthListeners.forEach(f => window.removeEventListener('message', f));
+  }
+
+  open(): void {
+    this.creatingNotebook = true;
+    this.loading = false;
+  }
+
+  close(): void {
+    this.creatingNotebook = false;
+  }
+
+  submitNewNotebook(): void {
+    const nbUrl = `/workspaces/` + encodeURIComponent(this.workspace.namespace) + '/' +
+        encodeURIComponent(this.workspace.id) + '/notebooks/create/?notebook-name=' +
+        encodeURIComponent(this.newName) + '&kernel-type=' +
+        encodeURIComponent(this.kernelType.toString());
+
+    const notebook = window.open(nbUrl, '_blank');
+
+    // TODO(RW-474): Remove the authHandler integration. This is messy,
+    // non-standard, and currently will break in the following situation:
+    // - User opens a new notebook tab.
+    // - While that tab is loading, user immediately navigates away from this
+    //   page.
+    // This is not easily fixed without leaking listeners outside the lifespan
+    // of the workspace component.
+    const authHandler = (e: MessageEvent) => {
```

Comment:
I'd complain about duplication here, but this code should be gone in the next couple of sprints so I don't think it's worth doing extra work.